lib/grammars.coffee

```diff
@@ -137,7 +137,10 @@ module.exports =
   Go:
     "File Based":
       command: "go"
-      args: (context) -> ['run', context.filepath]
+      args: (context) ->
+        if context.filepath.match(/_test.go/) then ['test', '' ]
+        else ['run', context.filepath]
+      workingDirectory: atom.workspace.getActivePaneItem()?.buffer?.file?.getParent?().getPath?()
```

Comment:
Ah, right. And for some languages that makes sense while in others it doesn't. :-/